[docs] added docs for jaeger integrations (#16693)
yugabyte/yugabyte-db: docs/content/preview/integrations/jaeger.md

- Root cause analysis
- Service dependency analysis
- Performance / latency optimization

Jaeger can be used with various storage backends. YugabyteDB's YCQL API can also be used as a storage backend for Jaeger (version 1.43.0, previous versions are unsupported).

The Jaeger example on this page uses a [Sample App: HotROD](https://www.jaegertracing.io/docs/1.43/getting-started/#sample-app-hotrod) to illustrate the usage of YCQL as a storage backend for jaeger.

Additionally, it uses Jaeger [All-in-one](https://www.jaegertracing.io/docs/1.43/deployment/#all-in-one), which is a special distribution that combines three Jaeger components: agent, collector, and query service/UI, in a single binary or container image.

{{< note title = "Note">}}

Following steps are for testing purpose only, for production deployment refer to [Jaeger documentation](https://www.jaegertracing.io/docs/1.18/).

{{< /note >}}

## Prerequisites

To use Jaeger, ensure that you have the following:

- YugabyteDB up and running. Download and install YugabyteDB by following the steps in [Quick start](../../quick-start/).

- [Go 1.20](https://go.dev/doc/go1.20).

## Build the application

> 

Reviewer: To build the application, do the following:

1. Jaeger natively supports two databases, Cassandra and Elasticsearch. So you need to set an environment variable `SPAN_STORAGE_TYPE` to specify your choice of database. For this example, set it to cassandra as follows:

    ```sh
    export SPAN_STORAGE_TYPE=cassandra
    ```

1. Clone the Jaeger repository.

    ```sh
    git clone git@github.com:jaegertracing/jaeger.git && cd jaeger
    ```

1. [Initialize](https://www.jaegertracing.io/docs/1.43/deployment/#schema-script) the cassandra keyspace and create the schema on YCQL using ycqlsh shell as follows:

    ```sh
    MODE=test sh ./plugin/storage/cassandra/schema/create.sh | path/to/ycqlsh
    ```

1. Verify that a keyspace `jaeger_v1_test` is created using ycqlsh as follows:

    ```sql
    ycqlsh> desc keyspaces;
    ```

    ```output
    system_auth  jaeger_v1_test  system_schema  system
    ```

1. From the jaeger project home directory, add the `jaeger-ui` submodule as follows:

    ```sh
    git submodule update --init --recursive
    ```

1. Compile, package the UI assets in the `jaeger-ui` source code, and run Jaeger All-in-one using the following command:

    ```sh
    make run-all-in-one
    ```

>     Wait for the logs to settle and navigate to <http://localhost:16686> to access the Jaeger UI which is used to go through the traces created by the application. This means that the Jaeger UI, collector, query, and agent are running.

Reviewer: ```suggestion
    Wait for the logs to settle, then navigate to <http://localhost:16686> to access the Jaeger UI, where you can view the traces created by the application. This indicates that the Jaeger UI, collector, query, and agent are running.
```

## Run the application

> 

Reviewer: To run the application, do the following:

1. From a new terminal, set the `SPAN_STORAGE_TYPE` environment variable again as follows:

    ```sh
    export SPAN_STORAGE_TYPE=cassandra
    ```

1. From your Jaeger project's home directory, start the HotROD application as follows:

    ```go
    go run ./examples/hotrod/main.go all
    ```

## Verify the integration

1. Navigate to the [landing page of HotROD application](http://localhost:8080) and click a few options on the page to verify if you get a response about a driver arriving in a few minutes. This will trigger a few traces which gets stored in the YCQL backend storage, and these traces can also be examined on Jaeger UI at <http://localhost:16686>.

> 1. You can also verify the integration from your ycqlsh shell using the following commands:

Reviewer: ```suggestion
You can also verify the integration from your ycqlsh shell using the following commands:
```

---
title: Jaeger
linkTitle: Jaeger
description: Use Jaeger with YCQL API
menu:
  preview_integrations:
    identifier: jaeger
    parent: integrations
    weight: 571
type: docs
---

Jaeger, inspired by [Dapper](https://research.google.com/pubs/pub36356.html) and [OpenZipkin](http://zipkin.io/), is a distributed tracing system released as open source by [Uber Technologies](http://uber.github.io/). It is used for monitoring and troubleshooting microservices-based distributed systems, including:

- Distributed context propagation
- Distributed transaction monitoring
- Root cause analysis
- Service dependency analysis
- Performance / latency optimization

> Jaeger can be used with various storage backends. YugabyteDB's YCQL API can also be used as a storage backend for Jaeger (version 1.43.0, previous versions are unsupported).

Reviewer 2: Can we add link to https://docs.yugabyte.com/preview/api/ycql/ where we mention YCQL.

The Jaeger example on this page uses the [HotROD](https://www.jaegertracing.io/docs/1.43/getting-started/#sample-app-hotrod) sample application to illustrate the use of YCQL as a storage backend.

Additionally, it uses Jaeger [All-in-one](https://www.jaegertracing.io/docs/1.43/deployment/#all-in-one), which is a special distribution that combines three Jaeger components, agent, collector, and query service/UI, in a single binary or container image.

{{< note title = "Note">}}

The following example is for testing purposes only. For production deployment, refer to the [Jaeger documentation](https://www.jaegertracing.io/docs/1.18/).

{{< /note >}}

## Prerequisites

To use Jaeger, ensure that you have the following:

- YugabyteDB up and running. Download and install YugabyteDB by following the steps in [Quick start](../../quick-start/).

- [Go 1.20](https://go.dev/doc/go1.20).

## Build the application

To build the application, do the following:

> 1. Jaeger natively supports two databases, Cassandra and Elasticsearch. So you need to set an environment variable `SPAN_STORAGE_TYPE` to specify your choice of database. For this example, set it to cassandra as follows:

Reviewer 2: Can we rephrase `For this example, set it to cassandra as follows:` because we need to set  `SPAN_STORAGE_TYPE` to Cassandra always if we are using YCQL as a backend storage, not only for this example.
